packages/vm/src/runTx.ts

```diff
@@ -320,7 +324,27 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
   /*
    * Execute message
    */
-  const { value, data, to } = tx
+  let { value, data, to } = tx
+
+  // Sapphire decode call data
+  let format, aead
+  if (this.confidential && data && data.length > 0) {
+    const { format, body: envelopBody } = cbor.decode(data)
+    if (format && format === 1) {
+      // X25519DeoxysII
+      const { nonce: deoxysiiNonce, data: envelopeData, pk } = envelopBody
+      const sharedKey = sha512_256.hmac
+        .create('MRAE_Box_Deoxys-II-256-128')
+        .update(nacl.scalarMult(this.secretKey, pk))
+        .arrayBuffer()
+      aead = new deoxysii.AEAD(new Uint8Array(sharedKey))
+      const { body } = cbor.decode(aead.decrypt(deoxysiiNonce, envelopeData))
+      data = Buffer.from(body.buffer)
+    } else {
+      // PLAIN
+      data = Buffer.from(envelopBody.buffer)
+    }
```

Comment:
Please use the corresponding `Cipher` from `@oasisprotocol/sapphire-paratime` since it already implements this logic. You won't need the deoxysii dep any more, also.